lib/logstash/filters/aggregate.rb

```diff
@@ -226,9 +248,15 @@ class LogStash::Filters::Aggregate < LogStash::Filters::Base
   # When this option is enabled, each time aggregate plugin detects a new task id, it pushes previous aggregate map as a new logstash event, 
   # and then creates a new empty map for the next task.
   # 
+  # When the timeout code is set, the event populated with the aggregation map is passed to the timeout code before the event is yielded
+  #
   # WARNING: this option works fine only if tasks come one after the other. It means : all task1 events, then all task2 events, etc...
   config :push_previous_map_as_event, :validate => :boolean, :required => false, :default => false
   
+
+  # When this option is enabled, each time the aggreagte flushes, the aggregate map is checked for timeouts. If a timeout is detected, 
+  # the plugin will write the aggregation map into a new event. This enables storing found tasks that never receive an end task   
+  config :push_map_as_event_on_timeout, :validate => :boolean, :required => false, :default => false
   
```

Comment:
I think we could make a better description, especially without explaining logstash plugin internals (like flush).
I suggest : 
When this option is enabled, each time a task timeout is detected, it pushes task aggregation map as a new logstash event.  
This enables to process task timeouts in logstash, but also to process tasks that have no explicit end event.

What you think ?